Lushu/src/main/kotlin/lushu/ContextGrammar/Grammar/Rules.kt

```diff
@@ -0,0 +1,210 @@
+package lushu.ContextGrammar.Grammar
+
+class Rules(private val root: Node = Node()) {
+
+    var line: MutableList<String> = mutableListOf()
+    var auxiliarLine: MutableList<String> = mutableListOf()
+    private fun isEndingPlusMatcher(current: Node?, index: Int): Boolean{
+
+        if(line[index].endsWith(logSeparator))
+            return true
+        else if(current == null){
+            return false
+        }
+        for(child in current!!.getChildren()){
+            if(child.match(line[index])){
+                if(child.isSensitive())
+                    auxiliarLine[index] = "*".repeat(auxiliarLine[index].length)
+                return true
+            }
+
+        }
+
+        return false
+    }
+    data class plusMatcherReturn(val i: Int, val node: Node?)
+    fun plusMatcher(node: Node?, initialIndex: Int): plusMatcherReturn{
+        var current: Node? = node ?: return plusMatcherReturn(1, null)
+
+        for(i in (initialIndex) until line.size){
+            if(isEndingPlusMatcher(current, i)){
+                line = auxiliarLine.toMutableList()
+                return plusMatcherReturn((i - initialIndex - 1), current)
+            }
+
+            else if((current == null) || !(current.match(line[i]))){
+                auxiliarLine = line.toMutableList()
+                return plusMatcherReturn(1, null)
+            }
+
+        }
+        println("WARNING: Unexpected behavior in plusMatcher function.")
+        auxiliarLine = line.toMutableList()
+        return plusMatcherReturn(1, null)
+    }
+
+    fun matcher(current: Node?, word: String): Node?{
+        if(current == null)
+            return null
```

Comment:
```suggestion
            return null
        val cur = current!!
```
You can use the `!!` operator only once here, I believe.